$NetBSD$

This message from swig is very mysterious; use sed(1) to patch that line.
    ifo_read.i:49: Error: Syntax error in input(1).
It reads "typedef int wchar_t;"

--- Makefile.orig	2006-05-02 19:17:13.000000000 +0000
+++ Makefile	2010-09-10 12:51:35.000000000 +0000
@@ -20,10 +20,10 @@
 PLINC      = $(shell perl -e \
 	       'use Config; print "$$Config{archlibexp}/CORE\n";')
 PLARCH     = $(shell perl -e 'use Config; print "$$Config{archname}\n";')
-PLCFLAGS   = $(shell perl -e 'use Config; print "$$Config{ccflags}\n";')
+PLCFLAGS   = $(shell perl -e 'use Config; print "$$Config{ccflags} $$Config{cccdlflags}\n";')
 
 CC         = gcc
-CFLAGS    ?= -pipe -O2 -fomit-frame-pointer -fpic
+CFLAGS     = -pipe -O2 -fomit-frame-pointer
 CFLAGS    += -I $(PLINC) $(PLCFLAGS) -Wall -W -Wno-strict-aliasing \
 	       -Wno-unused-function -Wno-unused-variable \
 	       -Wno-unused-parameter
@@ -31,7 +31,7 @@
 LOADLIBES ?= -ldvdread
 
 SWIG      ?= swig
-SWIGFLAGS ?= -perl -Wall -nodefault
+SWIGFLAGS ?= -perl -nodefaultctor -nodefaultdtor -D'__asm(x)='
 
 PUIC      ?= puic
 PUICFLAGS ?= -p 2
@@ -67,7 +67,7 @@
 		@rm -f $@.tmp
 
 %.msgs:		%.pm
-		./fakelupd.pl $* $^ > $@
+		perl ./fakelupd.pl $* $^ > $@
 
 %.qm:		%.ts
 		$(LRELEASE) -verbose $< -qm $@
@@ -84,10 +84,10 @@
 		$(CC) $(SOLDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
 
 libdvdread_wrap.c LibDvdRead.pm:	libdvdread.i ifo_read.i ignores.i
-		$(SWIG) $(SWIGFLAGS) $<
+		-$(SWIG) $(SWIGFLAGS) $<
 
 ifo_read.i:	ifo_read.c
-		$(CC) -E -P $< > $@
+		$(CC) -E -P $< | sed -e 's/typedef int wchar_t;//' > $@ 
 
 # .pm from .png
 DesignerMimeSourceFactory_qvamps.pm:	$(addprefix images/,$(IMAGES))
@@ -122,23 +122,24 @@
 
 # other targets go below
 install:	all
-		$(INSTALL) -m 755 -d $(PREFIX)/share/qvamps/etc \
-		  $(PREFIX)/share/qvamps/lib \
-		  $(PREFIX)/share/qvamps/translations \
-		  $(PREFIX)/lib/qvamps/$(PLARCH)
-		$(INSTALL) -m 755 qvamps $(PREFIX)/bin
+		$(INSTALL) -m 755 -d \
+		  $(DESTDIR)$(PREFIX)/bin \
+		  $(DESTDIR)$(PREFIX)/share/qvamps/lib \
+		  $(DESTDIR)$(PREFIX)/share/qvamps/translations \
+		  $(DESTDIR)$(PREFIX)/lib/qvamps/$(PLARCH)
+		$(INSTALL) -m 755 qvamps $(DESTDIR)$(PREFIX)/bin
 		$(INSTALL) -m 644 $(addsuffix .pm,$(UIMODULES) \
 		  $(MSGMODULES) $(GENMODULES) $(OTHMODULES)) \
-		  $(PREFIX)/share/qvamps/lib
+		  $(DESTDIR)$(PREFIX)/share/qvamps/lib
 		$(INSTALL) -m 644 $(addsuffix .qm,$(LANGUAGES)) \
-		  $(PREFIX)/share/qvamps/translations
+		  $(DESTDIR)$(PREFIX)/share/qvamps/translations
 		$(INSTALL) -m 644 penguin.jpg silence.mpa penguin_ntsc.mpg \
-		  penguin_pal.mpg $(PREFIX)/share/qvamps
-		$(INSTALL) -m 755 LibDvdRead.so $(PREFIX)/lib/qvamps/$(PLARCH)
+		  penguin_pal.mpg $(DESTDIR)$(PREFIX)/share/qvamps
+		$(INSTALL) -m 755 LibDvdRead.so $(DESTDIR)$(PREFIX)/lib/qvamps/$(PLARCH)
 
 uninstall:
-		rm -rf $(PREFIX)/bin/qvamps $(PREFIX)/share/qvamps \
-		  $(PREFIX)/lib/qvamps
+		rm -rf $(DESTDIR)$(PREFIX)/bin/qvamps $(DESTDIR)$(PREFIX)/share/qvamps \
+		  $(DESTDIR)$(PREFIX)/lib/qvamps
 
 clean:
 		rm -f LibDvdRead.so libdvdread_wrap.c ifo_read.i $(addsuffix \
